### PR TITLE
fix(addon-docs): assume links starting with "https://" are external

### PR DESCRIPTION
**Pull request by GuilleDF**
Issue: #17308

## What I did

- I updated the check for MDX links to see if the link starts with `https://`. Not all external links will, but this seems like a pretty easy way to detect most cases.
- I set `target="_blank"` as default for external links, but it can be overridden by the user. Let me know if you agree with this, or would prefer not to have this.

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by GuilleDF** [comment]
Not sure about the test failures, they do seem to pass on my local machine

File: addons/docs/src/blocks/mdx.tsx [hunk 1]
```diff
@@ -111,7 +111,7 @@ export const AnchorMdx: FC<AnchorMdxProps> = (props) => {
   }
 
   // External URL dont need any modification.
-  return <A {...props} />;
+  return <A target="_blank" {...props} />;
```
Comment (kylegach) [inline addons/docs/src/blocks/mdx.tsx:114]: There are two, separate changes in this PR.

The first, on line 94, updates the "is it an internal link?" heuristic in a helpful manner.

But this change coerces all external links to open in a new window, which we would consider a breaking change. If an author wants the link to open externally they can either continue adding `target="_blank"` to their links or, using the same override mechanism that would enable them to opt out of this change, they can opt _in_ instead.

In other words, please revert this part of your PR, and the first part is great.

**Review by kylegach — CHANGES_REQUESTED** [review]
Hi, @GuilleDF. Thanks for your PR and congratulations on your first contribution! I left a detailed comment which needs addressed, then this should be good-to-go.

File: addons/docs/src/blocks/mdx.tsx (hunk 1 above)
Comment (GuilleDF) [reply addons/docs/src/blocks/mdx.tsx]: > But this change coerces all external links to open in a new window, which we would consider a breaking change.

A counterpoint to this is that because of the previous heuristic, the only way a link would be considered external is if it had `target="_blank"`, which is why I thought adding this would keep the behavior as similar as possible. In my mind, this isn't a breaking change since before this there was no way to have an external link that _didn't_ open on a new window.

However, I do think it's an opinionated change, so I totally understand if you prefer to have it opt-in rather than opt-out. I've reverted it already! 😄

Comment (kylegach) [reply addons/docs/src/blocks/mdx.tsx]: That's a good point. You're absolutely right that it's an "opinionated change", not a "breaking" one. Thanks for understanding the intent, even when I didn't communicate clearly!

**Review by kylegach — APPROVED** [review]
This is great! Thanks again, @GuilleDF.
